### DISCUSSION TITLE
GH discussions : Launching a test :smile:

### DISCUSSION THREAD

**Discussion by 0penBrain** [FreeCAD/FreeCAD#9516] [Announcements]
@FreeCAD/maintainers & contributors : Following our developer meeting yesterday, I set up a minimal GH discussion organization so we can test how good or bad it can be for us to use it. :wink:

There are 2 sections:
* Maintainers section: only maintainers (and admins) can write in there. There are 2 categories ATM, one for announcements and one for discussions.
* Contributors section: it's a "public" section with only one category. Despite being "public", only technical discussions about coding (architecture, coding rules, PR, ...) are intended to happen there. Other discussions shall go to the forum.

**Comment by yorikvanhavre** [comment]
This looks good... I like being able to write *markdown* :sweat_smile:

**Comment by sliptonic** [comment]
Indeed.  And being able to create an issue from a discussion or reference a discussion in a new issue is handy.

**Comment by chennes** [comment]
I bet search is more functional, too.

**Comment by abdullahtahiriyo** [comment]
I will keep tuned.

**Comment by sliptonic** [comment]
Any thoughts about what kinds of subjects are best handled as discussions here vs forum topics?

Examples would be good

**Reply by 0penBrain** [reply]
> Any thoughts about what kinds of subjects are best handled as discussions here vs forum topics?
> 
> Examples would be good

Probably what would have previously fit in the "Developer corner" is better happening here.

**Comment by chennes** [comment]
Things where we want to talk about specific lines of code seem like an obvious fit, since it's so easy to start from a line of code and say "Reference in a discussion". As far as I am concerned, anything that requires code of any sort is nicer here, because Markdown is so much more pleasant than BBCode.

**Reply by sliptonic** [reply]
That's interesting since Github's own intro says this about discussions (emphasis mine).
>Discussions are for conversations that need to be transparent and accessible but do not need to be tracked on a project board and **are not related to code**, unlike GitHub Issues. Discussions enable fluid, open conversation in a public forum.

What about bigger topics that need more expert conversation  eg the need for a Materials workbench ?

**Reply by xtemp09** [reply]
> As far as I am concerned, anything that requires code of any sort is nicer here, because Markdown is so much more pleasant than BBCode.

I would disagree with you. Github markdown is a stub of BBCode. BBCode can highlight different [code] syntax, has [spoiler]s, character shortcuts, headlines, horizontal lines, tables, subscript, superscript and other pleasantries.[^1]

The forum's BBCode just needs update. Who maintains the forum?

[^1]: https://www.bbcode.org/reference.php

**Reply by yorikvanhavre** [reply]
me and @kkremitzki basically. But I am not sure you can manually replace or upgrade the bbcode of phpbb...

**Reply by chennes** [reply]
Two points: first, I said that Markdown was more pleasant (obviously a personal opinion), not that it has more features. Second, quite a few of those features you list are indeed available in Markdown (language selectors, tables, and headlines), and most use-cases for superscript I think are covered by using LaTeX (https://github.blog/changelog/2022-05-19-render-mathematical-expressions-in-markdown/).

## A headline

| First Header  | Second Header |
| ------------- | ------------- |
| Content Cell  | Content Cell  |
| Content Cell  | Content Cell  |

**The Cauchy-Schwarz Inequality**
$$\left( \sum_{k=1}^n a_k b_k \right)^2 \leq \left( \sum_{k=1}^n a_k^2 \right) \left( \sum_{k=1}^n b_k^2 \right)$$

**Reply by sliptonic** [reply]
Not to mention markdown opens the door for [mermaid](https://mermaid.js) like syntax that lets you have images, flowcharts, and other diagrams that are line based.

```mermaid
flowchart TD
    A[Start] --> B{Is it?}
    B -->|Yes| C[OK]
    C --> D[Rethink]
    D --> B
    B ---->|No| E[End]
```
